internal/services/loganalytics/log_analytics_workspace_resource.go

```diff
@@ -438,6 +451,17 @@ func resourceLogAnalyticsWorkspaceRead(d *pluginsdk.ResourceData, meta interface
 			d.Set("allow_resource_only_permissions", allowResourceOnlyPermissions)
 			d.Set("local_authentication_disabled", disableLocalAuth)
 
+			defaultDataCollectionRuleResourceId := ""
+			if props.DefaultDataCollectionRuleResourceId != nil {
+				dataCollectionId, err := datacollectionrules.ParseDataCollectionRuleID(*props.DefaultDataCollectionRuleResourceId)
+				if err != nil {
+					return fmt.Errorf("parsing %s: %+v", *props.DefaultDataCollectionRuleResourceId, err)
```

Comment:
Don't need to wrap this, the parser already returns sufficient information.
```suggestion
					return err
```